contracts/Aavegotchi/facets/ERC721BuyOrderFacet.sol

```diff
@@ -0,0 +1,190 @@
+    }
+
+    function cancelERC721BuyOrder(uint256 _buyOrderId) external {
+        address sender = LibMeta.msgSender();
+        ERC721BuyOrder memory erc721BuyOrder = s.erc721BuyOrders[_buyOrderId];
+
+        require(erc721BuyOrder.timeCreated != 0, "ERC721BuyOrder: ERC721 buyOrder does not exist");
+        require(
+            (sender == s.aavegotchis[erc721BuyOrder.erc721TokenId].owner) || (sender == erc721BuyOrder.buyer),
+            "ERC721BuyOrder: Only aavegotchi owner or buyer can call this function"
+        );
+        require((erc721BuyOrder.cancelled == false) && (erc721BuyOrder.timePurchased == 0), "ERC721BuyOrder: Already processed");
+
+        LibBuyOrder.cancelERC721BuyOrder(_buyOrderId);
+    }
+
+    function executeERC721BuyOrder(uint256 _buyOrderId) external {
+        address sender = LibMeta.msgSender();
+        ERC721BuyOrder memory erc721BuyOrder = s.erc721BuyOrders[_buyOrderId];
+
+        require(erc721BuyOrder.timeCreated != 0, "ERC721BuyOrder: ERC721 buyOrder does not exist");
+        require(sender == s.aavegotchis[erc721BuyOrder.erc721TokenId].owner, "ERC721BuyOrder: Only aavegotchi owner can call this function");
+        require((erc721BuyOrder.cancelled == false) && (erc721BuyOrder.timePurchased == 0), "ERC721BuyOrder: Already processed");
+
+        // disable for gotchi in lending
+        uint256 category = LibAavegotchi.getERC721Category(erc721BuyOrder.erc721TokenAddress, erc721BuyOrder.erc721TokenId);
+        if (category == LibAavegotchi.STATUS_AAVEGOTCHI) {
+            uint32 listingId = s.aavegotchiToListingId[uint32(erc721BuyOrder.erc721TokenId)];
+            require((listingId == 0) || (s.gotchiLendings[listingId].timeAgreed == 0), "ERC721BuyOrder: Not supported for aavegotchi in lending");
+        }
+
+        // hash validation
+        require(
+            erc721BuyOrder.validationHash ==
+                LibBuyOrder.generateValidationHash(erc721BuyOrder.erc721TokenAddress, erc721BuyOrder.erc721TokenId, erc721BuyOrder.validationOptions),
+            "ERC721BuyOrder: Invalid buy order"
+        );
+
+        erc721BuyOrder.timePurchased = block.timestamp;
+
+        LibERC721Marketplace.cancelERC721Listing(erc721BuyOrder.erc721TokenAddress, erc721BuyOrder.erc721TokenId, sender);
+
+        uint256 daoShare = erc721BuyOrder.priceInWei / 100;
+        uint256 pixelCraftShare = (erc721BuyOrder.priceInWei * 2) / 100;
+        //AGIP6 adds on 0.5%
+        uint256 playerRewardsShare = erc721BuyOrder.priceInWei / 200;
+
+        uint256 transferAmount = erc721BuyOrder.priceInWei - (daoShare + pixelCraftShare + playerRewardsShare);
+        LibERC20.transfer(s.ghstContract, s.pixelCraft, pixelCraftShare);
+        LibERC20.transfer(s.ghstContract, s.daoTreasury, daoShare);
+        LibERC20.transfer(s.ghstContract, sender, transferAmount);
+        //AGIP6 adds on 0.5%
+        LibERC20.transfer((s.ghstContract), s.rarityFarming, playerRewardsShare);
```

Comment:
Don't we have a function in LibMarketplace that handles this logic already?

Reply:
@orionstardust

Reply:
Update these.